module/sound/SpeechRecognition/src/SpeechRecognition.cpp

```diff
@@ -0,0 +1,39 @@
+#include "SpeechRecognition.hpp"
+
+#include <vosk_api.h>
+
+#include "extension/Configuration.hpp"
+
+#include "message/input/AudioData.hpp"
+
+namespace module::sound {
+
+    using extension::Configuration;
+    using message::input::AudioData;
+
+
+    SpeechRecognition::SpeechRecognition(std::unique_ptr<NUClear::Environment> environment)
+        : Reactor(std::move(environment)), config{} {
+
+        on<Configuration>("SpeechRecognition.yaml").then([this](const Configuration& cfg) {
+            // Use configuration here from file SpeechRecognition.yaml
+            this->log_level = cfg["log_level"].as<NUClear::LogLevel>();
+        });
+
+        // This reactor receives audiodata from emission
+        on<Trigger<AudioData>>().then([this](const AudioData& audioData) {
+            VoskModel* model           = vosk_model_new("/home/nubots/NUbots/model");
+            VoskRecognizer* recognizer = vosk_recognizer_new(model, 16000.0);
```

Comment:
The sample rate shouldn't be hardcoded - it's probably something we should get from the AudioData message. And we should only call this line if the sample rate changes.

Reply:
Here make a shared/unique pointer with a deleter that uses the free command.